docs/how_to/users_api.rst

```diff
@@ -0,0 +1,352 @@
+      "goals": "",
+      "is_active": false,
+      "last_login": null,
+      "mailing_address": "",
+      "requires_parental_consent": true,
+      "secondary_email": null,
+      "secondary_email_enabled": null,
+      "year_of_birth": null,
+      "phone_number": null,
+      "activation_key": "0511129811d04152bdb3ae49fa64c0eb",
+      "pending_name_change": null
+   }
+
+
+**Create User**
+
+URL: ``/eox-core/api/v1/user/``
+
+Method: POST
+
+Body:
+
+.. code-block:: json
+
+   {
+      "username": "johndoe",
+      "email": "johndoe@example.com",
+      "fullname": "John Doe",
+      "password": "p@ssword"
+   }
+
+Response Example:
+
+.. code-block:: json
+
+   200 OK
+
+   {
+      "email": "johndoe@example.com",
+      "username": "johndoe",
+      "is_active": false,
+      "is_staff": false,
+      "is_superuser": false
+   }
+
+
+
+**Extra Profile Fields**
+
+The User API supports the use of extra and custom registration fields for both Create and Update operations. This allows for flexibility in managing user profiles with additional fields beyond the default ones, ensuring that tenants can extend user data as needed.
+
+**Creating custom registration fields**
+
+**Tenant settings**
+
+To add custom or extra registration fields for a specific tenant, you'll need to configure the following settings:
+
+Example: Adding a custom field
+------------------------------
+
+If, for example, we want to add the field ``Organization name``, we will have to do the following:
+
+1. Add the field name, ``org_name`` for example, to ``extended_profile_fields`` setting. This indicates that ``org_name`` will be saved as an extended profile field.
+
+   .. code-block:: json
+
+      "extended_profile_fields": [ "org_name" ]
+
+2. Add ``org_name`` to ``REGISTRATION_EXTRA_FIELDS`` setting, indicating whether the field is hidden, optional, or required:
+
+   .. code-block:: json
+
+      "REGISTRATION_EXTRA_FIELDS": {
+         "org_name": "required"
+      }
+
+   **Note on Hidden Fields**
+
+   Fields that are set as ``hidden`` in the configuration will not be visible in the registration form or user profile, and they **cannot be updated through the API**.
+
+   If you attempt to update a field that is marked as ``hidden`` using the API, the update will be ignored, and no changes will be applied to that field.
+
+3. Define the custom field by creating it as a dictionary inside the ``EDNX_CUSTOM_REGISTRATION_FIELDS`` setting. In this case, we are creating a text field for ``org_name``. You must specify at least the ``name``, ``type``, and ``label``:
+
+   .. code-block:: json
+
+      "EDNX_CUSTOM_REGISTRATION_FIELDS": [
+         {
+            "name": "org_name",
+            "type": "text",
+            "label": "Organization name"
+         }
+      ]
+
+Once the field is configured, it can be included in the body of both ``POST`` (to create a new user) and ``PATCH`` (to update an existing user) requests.
+
+**Types of Custom Fields**
```

Comment:
I checked the code and found that there are more types:
https://github.com/eduNEXT/eox-core/blob/5f4bcb83101d53eac377bc0e57daec7fdae40f22/eox_core/api/v1/serializers.py#L38
Do you think we should mention them? Since I see they are being created as a `CharField`:
https://github.com/eduNEXT/eox-core/blob/5f4bcb83101d53eac377bc0e57daec7fdae40f22/eox_core/api/v1/serializers.py#L161-L172

Reply:
Better to add them so that none are missing. Thanks